packages/flutter_tools/lib/src/commands/migrate_apply.dart

```diff
@@ -0,0 +1,148 @@
+class MigrateApplyCommand extends FlutterCommand {
+  MigrateApplyCommand({
+    bool verbose = false,
+    required this.logger,
+    required this.fileSystem,
+  }) : _verbose = verbose {
+    requiresPubspecYaml();
+    argParser.addOption(
+      'working-directory',
+      help: 'Specifies the custom migration working directory used to stage and edit proposed changes.',
+      valueHelp: 'path',
+    );
+    argParser.addFlag(
+      'force',
+      abbr: 'f',
+      help: 'Ignore unresolved merge conflicts and apply staged changes by force.',
+    );
+  }
+
+  final bool _verbose;
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  @override
+  final String name = 'apply';
+
+  @override
+  final String description = r'Accepts the changes produced by `$ flutter migrate start` and copies the changed files into your project files. All merge conflicts should be resolved before apply will complete successfully. If conflicts still exist, this command will print the remaining conflicted files.';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final FlutterProject flutterProject = FlutterProject.current();
+
+    if (!await gitRepoExists(flutterProject.directory.path, logger)) {
+      logger.printStatus('No git repo found. Please run in a project with an initialized git repo or initialize one with:');
+      MigrateUtils.printCommandText('git init', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final bool force = boolArg('force');
+
+    Directory workingDirectory = flutterProject.directory.childDirectory(kDefaultMigrateWorkingDirectoryName);
+    if (stringArg('working-directory') != null) {
+      workingDirectory = fileSystem.directory(stringArg('working-directory'));
+    }
+    if (!workingDirectory.existsSync()) {
+      logger.printStatus('No migration in progress. Please run:');
+      MigrateUtils.printCommandText('flutter migrate start', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile = MigrateManifest.getManifestFileFromDirectory(workingDirectory);
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+    if (!checkAndPrintMigrateStatus(manifest, workingDirectory, warnConflict: true, logger: logger) && !force) {
+      logger.printStatus('Conflicting files found. Resolve these conflicts and try again.');
+      logger.printStatus('Guided conflict resolution wizard:');
+      MigrateUtils.printCommandText('flutter migrate resolve-conflicts', logger);
+      return const FlutterCommandResult(ExitStatus.fail);
+    }
+
+    if (await hasUncommittedChanges(flutterProject.directory.path, logger) && !force) {
+      return const FlutterCommandResult(ExitStatus.fail);
```

Comment:
Log to STDERR here why we're failing. Also, I think we should log something if they had uncommitted changes and we're overriding with `--force`